flink-metrics/flink-metrics-otel/src/test/java/org/apache/flink/metrics/otel/OpenTelemetryTestBase.java

```diff
@@ -0,0 +1,187 @@
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Tests for {@link OpenTelemetryMetricReporter}. */
+@ExtendWith(TestLoggerExtension.class)
+public class OpenTelemetryTestBase {
+    public static final Logger LOG = LoggerFactory.getLogger(OpenTelemetryTestBase.class);
+
+    private static final Duration TIME_OUT = Duration.ofMinutes(2);
+
+    @RegisterExtension
+    @Order(1)
+    private static final AllCallbackWrapper<TestContainerExtension<OtelTestContainer>>
+            OTEL_EXTENSION =
+                    new AllCallbackWrapper<>(
+                            new TestContainerExtension<>(() -> new OtelTestContainer()));
+
+    @BeforeEach
+    public void setup() {
+        Slf4jLevelLogConsumer logConsumer = new Slf4jLevelLogConsumer(LOG);
+        OTEL_EXTENSION.getCustomExtension().getTestContainer().followOutput(logConsumer);
+    }
+
+    public static OtelTestContainer getOtelContainer() {
+        return OTEL_EXTENSION.getCustomExtension().getTestContainer();
+    }
+
+    public static MetricConfig createMetricConfig() {
+        MetricConfig metricConfig = new MetricConfig();
+        metricConfig.setProperty(
+                OpenTelemetryReporterOptions.EXPORTER_ENDPOINT.key(),
+                getOtelContainer().getGrpcEndpoint());
+        return metricConfig;
+    }
+
+    public static void eventuallyConsumeJson(ThrowingConsumer<JsonNode, Exception> jsonConsumer)
+            throws Exception {
+        eventually(
+                () -> {
+                    // otel-collector dumps every report in a new line, so in order to re-use the
+                    // same collector across multiple tests, let's read only the last line
+                    getOtelContainer()
+                            .copyFileFromContainer(
+                                    getOtelContainer().getOutputLogPath().toString(),
+                                    inputStream -> {
+                                        List<String> lines = new ArrayList<>();
+                                        BufferedReader input =
+                                                new BufferedReader(
+                                                        new InputStreamReader(inputStream));
+                                        String last = "";
+                                        String line;
+
+                                        while ((line = input.readLine()) != null) {
+                                            lines.add(line);
+                                            last = line;
+                                        }
+
+                                        ObjectMapper mapper = new ObjectMapper();
+                                        JsonNode json = mapper.readValue(last, JsonNode.class);
+                                        try {
+                                            jsonConsumer.accept(json);
```

Comment:
What happens if there were no reports (and `last` == `""`)?
Will `readValue` throw an exception?

Reply:
Yes. But does it matter either way?